javaparser-core-testing/src/test/java/com/github/javaparser/ast/validator/Java16ValidatorTest.java

```diff
@@ -57,7 +57,7 @@ void yieldAllowed() {
     class Record {
 
         @Nested
-        class RecordAsIdentifierForbidden {
+        class RecordAsIdentifierSometimesForbidden {
```

Comment:
Additionally, you have correctly pointed out to me that records used as e.g. field and method identifiers is fine -- move the `recordUsedAsFieldName()` test method outside of this?